src/uagents/context.py

```diff
@@ -112,6 +130,29 @@ async def send(
             timeout=timeout,
         )
 
+    async def broadcast(
+        self,
+        destination: str,
+        message: Model,
+        timeout: Optional[int] = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
+    ):
```

Comment:
```suggestion
    async def experimental_broadcast(
        self,
        destination_protocol: str,
        message: Model,
        limit: Optional[int] = DEFAULT_SEARCH_LIMIT,
        timeout: Optional[int] = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
    ):
```

A few things:

* Parameter naming to make it more explicit.
* Related to the above we need to have a limit in the function
* Lastly, I think that we need some way to mark new functions as experimental or to mark that they are not part of the stable API (yet). I am not sure if this broadcast pattern is something that longer term that we will need, however, I think if we mark it as experimental then we can hedge our bets a little. @jrriehl / @lrahmani do you agree?

Reply:
This could be addressed by retaining `experimental_broadcast` for some deprecation period as a simple redirect to `broadcast`.